### ISSUE TITLE
Add more loss functions

### ISSUE THREAD

**Issue by bnubald** [CANARI-ML/canari-ml#27]
Alternate losses under consideration:
  - [x] HuberLoss (less sensitive to outliers than Mean Squared Error), ideally would need optimising of hyperparameter - costs would go up.
    - Less sensitive to outliers than Mean Squared Error Loss (i.e., better for this project due to spikes in wind speed).
    - Combines MSE and MAE, has a delta hyperparameter for tuning.
    - `delta` parameter controls transition between quadratic and linear behaviour.
    - Weight different regions, e.g. 20% in non-jet exit region, 80% in jet exit.
 
  - [ ] ~~Cosine Similarity~~
    - ~~(Captures directionality - couple of papers suggesting its performance is better with directional datasets such as Wind - not so well on scalars such as temperature)~~
  - [ ] Soft-DTW
    - Improves on above, includes aim to improve temporal stability.
    - Some variations of this look to be close to the state of the art.
